### PR TITLE
[Merged by Bors] - fix(TopologicalSpace/Basic): restore curly braces in `continuous_def`

### PR DESCRIPTION
**Pull request by grunweg**
These were placed on purpose; reverting them caused breakage downstream.

---

Follow-up to #9993.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Basic.lean [hunk 1]
```diff
@@ -1627,7 +1627,10 @@ scoped[Topology] notation (name := Continuous_of) "Continuous[" t₁ ", " t₂ "
 
 variable {f : X → Y} {s : Set X} {x : X} {y : Y}
 
-theorem continuous_def : Continuous f ↔ ∀ s, IsOpen s → IsOpen (f ⁻¹' s) :=
+-- The curly braces are intentional, so this definitions works well with simp
+-- when topologies are not those provided by instances.
+theorem continuous_def  {_ : TopologicalSpace X} {_ : TopologicalSpace Y} :
```
Comment (alreadydone) [inline Mathlib/Topology/Basic.lean:1632]:
```suggestion
theorem continuous_def {_ : TopologicalSpace X} {_ : TopologicalSpace Y} :
```

**Review by dwrensha — APPROVED** [review]

File: Mathlib/Topology/Basic.lean [hunk 2]
```diff
@@ -1627,7 +1627,10 @@ scoped[Topology] notation (name := Continuous_of) "Continuous[" t₁ ", " t₂ "
 
 variable {f : X → Y} {s : Set X} {x : X} {y : Y}
 
-theorem continuous_def : Continuous f ↔ ∀ s, IsOpen s → IsOpen (f ⁻¹' s) :=
+-- The curly braces are intentional, so this definitions works well with simp
+-- when topologies are not those provided by instances.
+theorem continuous_def {_ : TopologicalSpace X} {_ : TopologicalSpace Y} :
```
Comment (dwrensha) [inline Mathlib/Topology/Basic.lean:1632]: I don't know whether this is necessarily the right thing to do (my downstream code does not care one way or the other), but I just wanted to note: to fully revert the change, you would need to re-add the `{f : X → Y}` parameter here after the `TopologicalSpace` parameters. Then I think you would not need the other changes in `Order.lean`.
https://github.com/leanprover-community/mathlib4/blob/e7170d39769f41a993b9e30c9eff0f80c3fac97d/Mathlib/Topology/Basic.lean#L1631-L1632

Comment (sgouezel) [reply Mathlib/Topology/Basic.lean:1632]: Indeed, I think it's better to have `f` come last.

Comment (grunweg) [reply Mathlib/Topology/Basic.lean]: Fair point. I have changed the order.

**Review by dwrensha — APPROVED** [review]

File: Mathlib/Topology/Basic.lean [hunk 3]
```diff
@@ -1627,7 +1627,10 @@ scoped[Topology] notation (name := Continuous_of) "Continuous[" t₁ ", " t₂ "
 
 variable {f : X → Y} {s : Set X} {x : X} {y : Y}
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Basic.lean:1628]: Maybe this line should now go below `continuous_def`?

**Comment by urkud** [comment]
I'm sorry for merging #9993 too soon.

**Comment by urkud** [comment]
bors d=Ruben-VandeVelde

**Comment by grunweg** [comment]
I have incorporated the suggestion and merged master.

File: Mathlib/Topology/Basic.lean [hunk 4]
```diff
@@ -1499,12 +1499,15 @@ variable {X Y Z : Type*} [TopologicalSpace X] [TopologicalSpace Y] [TopologicalS
 
 open TopologicalSpace
 
-variable {f : X → Y} {s : Set X} {x : X} {y : Y}
-
-theorem continuous_def : Continuous f ↔ ∀ s, IsOpen s → IsOpen (f ⁻¹' s) :=
+-- The curly braces are intentional, so this definitions works well with simp
```
Comment (urkud) [inline Mathlib/Topology/Basic.lean:1502]:
```suggestion
-- The curly braces are intentional, so this definition works well with simp
```

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**